src/logger.service.ts

```diff
@@ -24,126 +29,138 @@ const Levels = [
+    } else {
+      return level;
+    }
   }
 
-  private _logOnServer(level: string, message: string) {
-    if (!this.options.serverLoggingUrl) { return; };
+  private _logOnServer(level: NgxLoggerLevel, message, additional: any[]) {
+    if (!this.options.serverLoggingUrl) {
+      return;
+    }
 
     // if the user provides a serverLogLevel and the current level is than that do not log
-    if (this._serverLogLevelIdx && Levels.indexOf(level) < this._serverLogLevelIdx) { return; };
+    if (level < this._serverLogLevel) {
+      return;
+    }
 
     const headers = new HttpHeaders().set('Content-Type', 'application/json');
 
-    this.http.post(this.options.serverLoggingUrl, { level: level, message: message, timestamp: this._timestamp() }, { headers: headers })
-      .catch(error => error)
+    this.http.post(this.options.serverLoggingUrl, {
+      level: level,
+      message: message,
+      additional: additional,
+      timestamp: this._timestamp()
+    }, {headers})
       .subscribe(
-      res => null,
-      error => this._log('ERROR', 'FAILED TO LOG ON SERVER', false)
+        res => null,
+        error => this._log(NgxLoggerLevel.ERROR, false, 'FAILED TO LOG ON SERVER')
       );
-
   }
 
-  private _logIE(level: string, message: any) {
-    message = `${this._timestamp()} [${level}] ${message}`;
-
-    if (level === 'WARN') {
-      console.warn(message);
-    } else if (level === 'ERROR') {
-      console.error(message);
-    } else if (level === 'INFO') {
-      console.info(message);
-    } else {
-      console.log(message);
+  private _logIE(level: NgxLoggerLevel, message: string, additional: any[]) {
+    switch (level) {
+      case NgxLoggerLevel.WARN:
+        console.warn(`${this._timestamp()} [${level}] `, message, ...additional);
+        break;
+      case NgxLoggerLevel.ERROR:
+        console.error(`${this._timestamp()} [${level}] `, message, ...additional);
+        break;
+      case NgxLoggerLevel.INFO:
+        console.info(`${this._timestamp()} [${level}] `, message, ...additional);
+        break;
+      default:
+        console.log(`${this._timestamp()} [${level}] `, message, ...additional);
     }
   }
 
-  private _log(level: string, message: any, logOnServer: boolean) {
+  private _log(level: NgxLoggerLevel, logOnServer: boolean, message, additional: any[] = []) {
 
     // if no message or the log level is less than the environ
-    if (!message || Levels.indexOf(level) < this._clientLogLevelIdx) { return; };
+    if (!message || level < this._clientLogLevel) {
+      return;
+    }
 
     if (logOnServer) {
-      this._logOnServer(level, message);
+      this._logOnServer(level, message, additional);
     }
 
     if (typeof message === 'object') {
       try {
         message = JSON.stringify(message, null, 2);
       } catch (e) {
-        message = `circular object in message: ${message}`;
+        message = 'circular object in message. ';
+        additional = [message, ...additional];
```

Comment:
@dbfannin Instead of "circular object in message: [Object object]", print "circular object in message" and print message as additional object.